GravySDK/src/main/java/com/gr4vy/android_sdk/models/Gr4vyBuyer.kt

```diff
@@ -0,0 +1,46 @@
+package com.gr4vy.android_sdk.models
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+import kotlinx.serialization.Serializable
+
+@Serializable
+@Parcelize
+data class Gr4vyBuyer(
+  val displayName: String? = null,
+  val externalIdentifier: String? = null,
+  val billingDetails: Gr4vyBillingShippingDetails? = null,
+  val shippingDetails: Gr4vyBillingShippingDetails? = null
```

Comment:
I would create a separate class for the `shipping-details` since they could differ in the future. Also `shipping-details` does not have a `taxId`.

Not an expert but does `taxId` get added to the request payload as `taxId: null` or its not added at all?

Reply:
If it's not defined in the schema it'll return a 400 due to extra fields not being permitted. Anyway, that won't be a problem if you separate both.

Reply:
It is ok for billingDetails, but for shippingDetails `core-api` will return a bad request.

<img width="537" alt="image" src="https://github.com/user-attachments/assets/a7d7b343-9337-4d06-a168-1b31fcaad991">

Reply:
I'm seeing this in the API response which looks ok (and I can see those details are empty in dashboard, only the ones I set are visible)

```
'buyer': {'id': None, 'type': 'buyer', 'display_name': None, 'external_identifier': None, 'billing_details': {'first_name': 'Ivy', 'last_name': 'Leaf', 'email_address': None, 'phone_number': None, 'address': {'city': 'London', 'country': None, 'postal_code': None, 'state': None, 'state_code': None, 'house_number_or_name': None, 'line1': None, 'line2': None, 'organization': None}, 'tax_id': None}}
```

Reply:
Yeah, it's okay for billing details

Reply:
updated, thanks guys